x-pack/plugins/enterprise_search/common/ml_inference_pipeline/index.test.ts

```diff
@@ -232,6 +232,43 @@ describe('generateMlInferencePipelineBody lib function', () => {
     );
   });
 
+  it('should return something that safely removes redundant prefixes', () => {
+    const mockTextClassificationModel: MlTrainedModelConfig = {
+      ...mockModel,
+      ...{ inference_config: { text_classification: {} } },
+    };
+    const actual: MlInferencePipeline = generateMlInferencePipelineBody({
+      description: 'my-description',
+      model: mockTextClassificationModel,
+      pipelineName: 'my-pipeline',
+      fieldMappings: [
+        { sourceField: 'my-source-field', targetField: 'ml.inference.my-source-field_expanded' },
+      ],
+    });
+
+    expect(actual).toEqual(
+      expect.objectContaining({
+        description: expect.any(String),
+        processors: expect.arrayContaining([
+          expect.objectContaining({
+            remove: {
+              field: 'ml.inference.my-source-field_expanded',
+              ignore_missing: true,
+            },
+          }),
+          expect.objectContaining({
+            inference: expect.objectContaining({
+              field_map: {
+                'my-source-field': 'MODEL_INPUT_FIELD',
+              },
+              target_field: 'ml.inference.my-source-field_expanded',
+            }),
+          }),
+        ]),
+      })
+    );
+  });
+
```

Comment:
worth adding a test to make sure that fields like `my-source-ml.inference-field`, `ml.inference.my-source-ml.inference-field_expanded` don't become `ml.inference.ml-source-field_expanded`?

Reply:
good call